src/EventPlanner.Application/Services/LocationService.cs

```diff
@@ -17,26 +18,49 @@ public LocationService(ILocationRepository locationRepo)
     /// </summary>
     public async Task<Location> CreateAsync(Location location)
     {
+       
+        if(location is null)
+            throw new ValidationException("Location is required");
+        if(string.IsNullOrWhiteSpace(location.Name))
+            throw new ValidationException("Location name is required");
+        if(string.IsNullOrWhiteSpace(location.Address))
+            throw new ValidationException("Location address is required");
+        if (location.Capacity <= 0)
+            throw new ValidationException("Location capacity must be greater than 0");
```

Comment:
You have same code to validate location entity inside CreateAsync and UpdateAsync, move validation logic to another place or read how to use FluentValidation